cfme/tests/automate/test_class.py

```diff
@@ -169,3 +170,33 @@ def test_class_display_name_unset_from_ui(get_namespace):
     with update(a_class):
         a_class.display_name = ""
     assert a_class.exists
+
+
+@pytest.mark.tier(3)
+def test_automate_schema_field_without_type(klass):
+    """It shouldn't be possible to add a field without specifying a type.
+
+    Polarion:
+        assignee: ghubale
+        casecomponent: Automate
+        caseimportance: medium
+        caseposneg: negative
+        initialEstimate: 1/12h
+        tags: automate
+        testSteps:
+            1. Create a schema field that does not specify a type
+            2. Save the schema
+        expectedResults:
+            1.
+            2. it is not possible to add a field that does not specify the type
+               (assertion, attribute, relationship, ...)
+
+    Bugzilla:
+        1365442
+    """
+    with pytest.raises(AssertionError):
+        klass.schema.add_fields({'name': 'execute', 'data_type': 'String'})
+    view = klass.create_view(ClassSchemaEditView)
+    assert view.schema.save_button.disabled
+    assert view.schema.reset_button.disabled
+    assert not view.schema.cancel_button.disabled
```

Comment:
Question: Is there an instance when the cancel button is disabled? Why do we need this assertion?

Reply:
@john-dupuy 
- If you are talking about automate ```instance``` then I would like to tell you that this is related to only ```schema```  of automate class.
-  So here in this test case, I just want to make sure that these buttons are in expected States while creating schema with no ```TYPE```

Reply:
@ganeshhubale I don't mean automate `instance`, I mean that I have never seen a case where the cancel button is disabled, since the user is always able to cancel their edits. Regardless, it doesn't hurt anything to have that assertion in there.

Reply:
@john-dupuy :100: 
@john-dupuy more assertion in automation; I like to accept. this is normal behavior that `cancel button never disable` but check is in positive side.